dropshot/examples/colliding-error-type-names.rs

```diff
@@ -0,0 +1,165 @@
+                .status(http::StatusCode::BAD_REQUEST)
+                .header("x-request-id", ctx.request_id)
+                .body(
+                    serde_json::to_string(self)
+                        .expect("serialization of MyError should never fail")
+                        .into(),
+                )
+                .expect("building response should never fail")
+        }
+    }
+
+    impl std::fmt::Display for Error {
+        fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+            match self {
+                Self::OverfullHbox { badness } => {
+                    write!(f, "overfull hbox, badness {}", badness)
+                }
+                Self::UnderfullHbox { badness } => {
+                    write!(f, "underfull hbox, badness {badness}")
+                }
+            }
+        }
+    }
+
+    #[endpoint {
+    method = GET,
+    path = "/pdflatex/{filename}",
+}]
+    pub(super) async fn get_pdflatex(
+        _rqctx: RequestContext<()>,
+        _path: dropshot::Path<PdflatexPathParams>,
+    ) -> Result<dropshot::HttpResponseCreated<Pdf>, Error> {
+        Err(Error::OverfullHbox { badness: 1000 })
+    }
+
+    #[derive(
+        Debug, serde::Deserialize, serde::Serialize, schemars::JsonSchema,
+    )]
+    struct PdflatexPathParams {
+        filename: String,
+    }
+
+    #[derive(
+        Debug, serde::Deserialize, serde::Serialize, schemars::JsonSchema,
+    )]
+    struct Pdf {
+        filename: String,
+    }
+}
+
+mod flask {
+    use super::*;
+    #[derive(
+        Debug, serde::Deserialize, serde::Serialize, schemars::JsonSchema,
+    )]
+    struct Flask {
+        name: String,
+    }
+    #[derive(
+        Debug, serde::Deserialize, serde::Serialize, schemars::JsonSchema,
+    )]
+    struct Error {
+        reason: String,
+    }
+
+    impl dropshot::error::IntoErrorResponse for Error {
+        fn into_error_response(
+            &self,
+            ctx: dropshot::error::ErrorContext<'_>,
+        ) -> http::Response<dropshot::Body> {
+            http::Response::builder()
+                .status(http::StatusCode::NOT_FOUND)
+                .header("x-request-id", ctx.request_id)
+                .body(
+                    serde_json::to_string(self)
+                        .expect("serialization of Error should never fail")
+                        .into(),
+                )
+                .expect("building response should never fail")
+        }
+    }
+
+    impl std::fmt::Display for Error {
+        fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+            f.write_str(self.reason.as_str())
+        }
+    }
+
+    /// Gets ye flask.
+    #[endpoint {
+        method = GET,
+        path = "/ye-flask",
+    }]
+    pub(crate) async fn get_ye_flask(
+        _rqctx: RequestContext<()>,
+    ) -> Result<dropshot::HttpResponseOk<Flask>, Error> {
+        Err(Error { reason: "can't get ye flask!".to_string() })
+    }
+}
```

Comment:
What is this example adding?

edit: I think I realized this one is trying to show what happens if you use the same error type name in multiple places?  I would have thought it'd just be a test case and not an example.  Is that something we'd expect a user to wonder how to do and consult this example for?

Reply:
This was really intended as a test case. As far as I could tell, there doesn't seem to be much infrastructure for testing that generated OpenAPI docs are well-formed, but I may have overlooked it. I wrote this as an example just to test this edge case but was hoping to turn it into a proper test before merging this.

Reply:
Makes sense but yeah there are definitely expectorate tests for generated OpenAPI docs (e.g., `test_openapi.rs`).